src/IMG_gif.c

```diff
@@ -2419,3 +2419,52 @@ bool IMG_CreateGIFAnimationStream(struct IMG_AnimationStream *stream, SDL_Proper
     return true;
 #endif /*!defined(SAVE_GIF)*/
 }
+
+#ifdef SAVE_GIF
+
+bool IMG_SaveGIF_IO(SDL_Surface* surface, SDL_IOStream* dst, bool closeio)
+{
+    IMG_AnimationStream* stream = IMG_CreateAnimationStream_IO(dst, closeio, "gif");
+    if (!stream) {
+        SDL_SetError("Failed to create GIF animation stream.");
+        return false;
```

Comment:
```suggestion
        return SDL_SetError("Failed to create GIF animation stream.");
```
SDL_SetError() returns a failure code. Can you change this here, and below?